listenbrainz/webserver/static/js/src/utils/APIService.ts

```diff
@@ -1172,4 +1172,26 @@ export default class APIService {
     await this.checkStatus(response);
     return response.json();
   };
+
+  fetchSitewideFreshReleases = async (
+    release_date?: string,
+    days?: number
+  ): Promise<any> => {
+    let url = `${this.APIBaseURI}/explore/fresh-releases/`;
+    if (!isUndefined(release_date)) url += `?release_date=${release_date}`;
+    if (!isUndefined(days)) url += `&days=${days}`;
```

Comment:
Not sure this is very safe; if release_date is undefined but not days, you end up with a malformed url `/explore/fresh-releases/&days=123` without  the required `?`.

Have a look at the implementation of query parameters in `getListensForUser` for example.

Reply:
Actually the request will go as `/explore/fresh-releases/?release_date=&days=123` (as it currently does).

But I agree that it's not the ideal query params implementation. I'll check out the `getListensForUser`.

Reply:
Update: url query params are safe now :)